hooks/pre_validations/30-upgrade_postgresql.rb

```diff
@@ -0,0 +1,32 @@
+def check_postgresql_storage
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  current_postgres_dir = '/var/lib/pgsql'
+  new_postgres_dir = '/var/opt/rh/rh-postgresql10/lib/pgsql'
+
+  begin
+    postgres_size = `du -b -s #{current_postgres_dir}`.split[0].to_i
+
+    if available_space(new_postgres_dir) < postgres_size
+      fail_and_exit "The PostgreSQL upgrade requires at least #{(postgres_size / 1024) / 1024} MB of storage."
```

Comment:
The message should probably tell the user *where* the storage is required as the user is not aware of the move to `/var/opt/rh/rh-postgresql10/lib/pgsql` just yet.